test/aws_credentials_providers_SUITE.erl

```diff
@@ -141,12 +141,17 @@ teardown_provider(Context) ->
 
 mock_httpc_request_ec2(Method, Request, HTTPOptions, Options, Profile) ->
   case Request of
-    {"http://169.254.169.254/latest/meta-data/iam/security-credentials/", []} ->
+    {"http://169.254.169.254/latest/api/token/", _RequestHeaders} ->
+      {ok, response('session-token')};
+    % TODO: Pattern match the session token in RequestHeaders below
```

Comment:
There may be multiple headers in `RequestHeaders`.  I'd like to just ensure that the session token header returned from the first call appears in the appropriate header for subsequent requests.  Still looking into how to do this properly.